utils/streams.js

```diff
@@ -0,0 +1,147 @@
+const extractFileData = (fileName) => {
+  const reader = fs.createReadStream(fileName);
+  reader.pipe(process.stdout);
+  reader.on('end', () => console.info('End of file'.blue))
+}
+
+// task 5
+const inputOutputNames = ["inputOutput", "input-output", "io"];
+const inputOutput = () => {
+  process.stdin
+    .pipe(through2(function (chunk, enc, cb) {
+      const str = chunk.toString();
+      const upperStr = str.toUpperCase();
+      this.push(upperStr);
+      cb();
+    }))
+    .pipe(process.stdout);
+}
+
+// task 6
+const csvToJSONConsoleNames = ["csvToJSONConsole", "csvToJSONOut", "csvJSON"];
+const csvToJSONOut = (fileName) => {
+  if (path.extname(fileName).toLowerCase() !== '.csv') return ;
+  fs.createReadStream(fileName)
+    .pipe(through2(function (chunk, enc, cb) {
+      const str = chunk.toString();
+      const json = csvToJSON(str);
+      this.push(json);
+      cb();
+    }))
+    .pipe(process.stdout);
+}
+
+// task 7
+const csvToJSONFileNames = ["csvToJSONFile", , "csvJSONFile"];
+const csvToJSONFile = (fileName) => {
+  const extName = path.extname(fileName);
+  if (extName.toLowerCase() !== '.csv') return ;
+
+  const baseName = path.basename(fileName, extName);
+  const pathName = path.dirname(fileName);
+  const writer = fs.createWriteStream(`${pathName}/${baseName}.json`);
+
+  writer.on("finish", () => console.info(`${baseName}.json was created`.bgRed.bgYellow));
+
+  fs.createReadStream(fileName)
+    .pipe(through2(function (chunk, enc, cb) {
+      const str = chunk.toString();
+      const json = csvToJSON(str);
+      const fixedJson = json.replace(/\s/g, "");
+      this.push(fixedJson);
+      cb();
+    }))
+    .pipe(writer);
+}
+
+// task 8
+const readdir = promisify(fs.readdir);
+const cssBundleNames = ["cssBundle", "cssBundler", "css", "bundler"];
+const cssBundle = (dirname) => {
+  const cssUrl = "https://www.epam.com/etc/clientlibs/foundation/main.min.fc69c13add6eae57cd247a91c7e26a15.css";
+  readdir(dirname)
+    .then((files) => {
+      if (!files || (files instanceof Array && files.length === 0)) {
+        console.info("No .css files to bundle".bgRed.white)
+        return ;
+      }
+
+      const writer = fs.createWriteStream(`${dirname}/bundle.css`, { flags: "a"});
+      files.map((file) => {
+        if (path.extname(file).toLowerCase() === ".css" && file !== "bundle.css") {
+          fs.createReadStream(`${dirname}/${file}`)
+            .pipe(through2(function(chunk, enc, cb) {
+              const css = chunk.toString() + "\n";
+              this.push(css);
+              cb();
+            }))
+            .pipe(writer);
+        }
+      });
+
+      const httpWriter = fs.createWriteStream(`${dirname}/bundle.css`, { flags: "a"});
+      httpWriter.on("finish", () => console.log("Closing writing".bgRed.white))
+      https.get(cssUrl, (res) => {
+        res.pipe(httpWriter);
+      });
+    })
+    .catch(err => console.log(err));
+}
+
```

Comment:
Wrap each task into a function which should be export. The last task/function should be default exported